src/app/account/login/login.component.ts

```diff
@@ -58,6 +67,35 @@ export class LoginComponent implements OnInit, OnDestroy {
             this.errorMessage = LOGIN_ERROR_MESSAGES[error] || LOGIN_ERROR_MESSAGES.internalError;
         }
     }
+    onSubmit() {
+        const encryptedPassword = this.authService.encryptPassword(this.loginGroup.value.password);
+        const loginData = {
+            'email': this.loginGroup.value.email,
+            'password': encryptedPassword,
+          };
+        this.disableLoginBtn = true;
+
+        this.authService.basicAuthLogin(loginData)
+        .pipe(finalize(() => { this.disableLoginBtn = false; }))
```

Comment:
It could be simplified further:
```suggestion
        .pipe(finalize(() => this.disableLoginBtn = false))
```